libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -69,20 +73,21 @@ instance
   type TypeSpec fn (Map k v) = MapSpec fn k v
   type Prerequisites fn (Map k v) = (HasSpec fn k, HasSpec fn v)
 
-  emptySpec = MapSpec mempty mempty mempty mempty NoFold
+  emptySpec = defaultMapSpec
 
   combineSpec
-    (MapSpec mustKeys mustVals size kvs foldSpec)
-    (MapSpec mustKeys' mustVals' size' kvs' foldSpec') = fromGE ErrorSpec $ do
+    (MapSpec mHint mustKeys mustVals size kvs foldSpec)
+    (MapSpec mHint' mustKeys' mustVals' size' kvs' foldSpec') = fromGE ErrorSpec $ do
       typeSpec
         . MapSpec
+          (unionWithMaybe min mHint mHint')
```

Comment:
Any particular reason to use 'min', lots of things might make sense (max, average)

Reply:
I added a comment:
```
          -- This is min because that allows more compositionality - if a spec specifies a
          -- low upper bound because some part of the spec will be slow it doesn't make sense
          -- to increase it somewhere else because that part isn't slow.
```